packages/lesswrong/components/form-components/FormComponentPostEditorTagging.tsx

```diff
@@ -19,6 +17,26 @@ const styles = (theme: ThemeType): JssStyles => ({
   },
 });
 
+/**
+ * Split a single array of tag ids into separate arrays based on boolean
+ * predicates
+ */
+const splitBy = (
```

Comment:
There is a lodash function `groupBy` which does a similar (though not exactly the same) thing fyi